### PR TITLE
CI: reuse caching steps

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
no idea why suddenly all tests in the complete testing takes so much longer while nothing changed on the package side and caching is still within expectation limit...

File: .github/workflows/ci_test-base.yml [hunk 1]
```diff
@@ -30,30 +33,17 @@ jobs:
...
-    # Note: This uses an internal pip API and may not always work
-    # https://github.com/actions/cache/blob/master/examples.md#multiple-oss-in-a-workflow
-    - name: Get pip cache
-      id: pip-cache
-      run: |
-        python -c "from pip._internal.locations import USER_CACHE_DIR; print('::set-output name=dir::' + USER_CACHE_DIR)"
-
-    - name: Cache pip
-      uses: actions/cache@v2
+    - uses: ./.github/actions/cashing
```
Comment (ethanwharris) [inline .github/workflows/ci_test-base.yml:42]: Should be `caching`(folder name will need to change too)
```suggestion
    - uses: ./.github/actions/caching
```

**Review by ethanwharris — COMMENTED** [review]
Just a small comment :smiley:

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Comment by akihironitta** [comment]
@Borda Is it working properly? I had a look at some runs, and it looks like it doesn't use caching at all if I see it correctly. Just curious if you have any idea why it's not working?

**Comment by Borda** [comment]
> @Borda Is it working properly? I had a look at some runs, and it looks like it doesn't use caching at all if I see it correctly. Just curious if you have any idea why it's not working?

could you point me to the case? I was testing it on HF cache and it seems to be restored, but still I could make a mistake :rabbit:

**Review by SkafteNicki — APPROVED** [review]

**Comment by akihironitta** [comment]
> > @Borda Is it working properly? I had a look at some runs, and it looks like it doesn't use caching at all if I see it correctly. Just curious if you have any idea why it's not working?
> 
> could you point me to the case? I was testing it on HF cache and it seems to be restored, but still I could make a mistake 🐰

I had a look again, and apparently, I was wrong😄 All seem to be working properly :)

**Comment by Borda** [comment]
> I had a look again, and apparently, I was wrong All seem to be working properly :)

not wrong yeah, at the time there was a bug that Caching does not create even an empty folder if the cache is empty, so it was fixed in meantime by soft folder creation...

File: .github/workflows/ci_test-full.yml [hunk 2]
```diff
@@ -80,46 +83,27 @@ jobs:
...
 
-    - name: Install dependencies
+    - name: Install base dependencies
       run: |
         pip --version
         pip install --requirement requirements.txt --upgrade --find-links $PYTORCH_URL
+
+    - name: Freeze Numpy (hotfix)
+      if: matrix.requires == 'latest'
+      run: pip install -q "numpy==1.20.0"  # temp fix for cypesq
```
Comment (akihironitta) [inline .github/workflows/ci_test-full.yml:99]: For the record, the numpy version was fixed due to #841.

**Pull request merged**